django/middleware/csrf.py

```diff
@@ -63,21 +101,28 @@ def rotate_token(request):
     """
     request.META.update({
         "CSRF_COOKIE_USED": True,
-        "CSRF_COOKIE": _get_new_csrf_key(),
+        "CSRF_COOKIE": _get_new_csrf_token(),
     })
 
 
 def _sanitize_token(token):
     # Allow only alphanum
-    if len(token) > CSRF_KEY_LENGTH:
-        return _get_new_csrf_key()
```

Comment:
Looking at the code changes below in the middleware, I also see no clear reason for not returning a new token here to stay a little bit more compatible to with what we had before.